test/acceptance/team_test.rb

```diff
@@ -57,4 +57,38 @@ def test_search_for_public_teams
       end
     end
   end
+
+  def test_team_pagination
+    user = User.create(username: "foobar", github_id: 123)
+
+    attributes = { slug: 'some-team', name: 'Some Team', confirmed: true }
+    team = Team.by(user).defined_with(attributes, user)
+    team.save!
+
+    TeamMembership.create!(team_id: team.id, user_id: user.id, confirmed: true)
+
+    submission = Submission.create(user: user,
+                                   language: 'ruby',
+                                   slug: 'leap',
+                                   solution: { 'leap.rb': 'CODE' },
+                                   created_at: Time.now - 10)
+
+    3.times do |index|
+      UserExercise.create(user: user,
+                          submissions: [submission],
+                          language: 'fake',
+                          slug: "apple#{index}",
+                          iteration_count: 1,
+                          last_activity_at: Time.now - 10,
+                          last_iteration_at: Time.now - 10,
+                          fetched_at: Time.now - 10,
+                          skipped_at: Time.now - 10)
```

Comment:
If the tests run quickly there might not be a noticable difference in the created at / updated at timestamps, which we use to sort the page.

This test could fail in a seemingly random way (sometimes passing and sometimes failing). I would suggest that you use the index of your loop in some way to make sure that the time difference is explicit.